x-pack/plugins/infra/public/store/local/log_flyout/actions.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import actionCreatorFactory from 'typescript-fsa';
+const actionCreator = actionCreatorFactory('x-pack/infra/local/log_flyout');
+export const setFlyoutItem = actionCreator<string>('SET_FLYOUT_ITEM');
+export const showFlyout = actionCreator<boolean>('SHOW_FLYOUT');
```

Comment:
In order to "make forbidden states unrepresentable", how about we turn these into

```typescript
export const hideFlyout = actionCreator('HIDE_FLYOUT');
export const showFlyout = actionCreator<string>('SHOW_FLYOUT');
```

? That way it's not possible to show the flyout without specifying an item id instead of relying on calling these always together and in the right order.